src/catalog/src/builtin.rs

```diff
@@ -2862,15 +2862,51 @@ LEFT JOIN mz_catalog.mz_databases d ON d.id = s.database_id
 WHERE s.database_id IS NULL OR d.name = current_database()",
 };
 
+// Note: Databases, Roles, Clusters, Cluster Replicas, Secrets, and Connections are excluded from
+// this view.
 pub const PG_DESCRIPTION: BuiltinView = BuiltinView {
     name: "pg_description",
     schema: PG_CATALOG_SCHEMA,
-    sql: "CREATE VIEW pg_catalog.pg_description AS SELECT
-    c.oid as objoid,
-    NULL::pg_catalog.oid as classoid,
-    0::pg_catalog.int4 as objsubid,
-    NULL::pg_catalog.text as description
-FROM pg_catalog.pg_class c",
+    sql: "CREATE VIEW pg_catalog.pg_description AS
+SELECT
+	pg_obj.objoid,
+	pg_obj.classoid,
+	COALESCE(cmt.object_sub_id::INT4, 0) AS objsubid,
+	cmt.comment AS description
+FROM
+	(
+		SELECT
+			oid, (SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_class')
+		FROM
+			pg_catalog.pg_class
+		UNION ALL
+			SELECT
+				oid,
+				(SELECT oid FROM mz_catalog.mz_objects WHERE name = 'pg_type')
+			FROM
+				pg_catalog.pg_type
+		UNION ALL
+			SELECT
+				oid,
+				(
+					SELECT
+						oid
+					FROM
+						mz_catalog.mz_objects
+					WHERE
+						name = 'pg_namespace'
+				)
+			FROM
+				pg_catalog.pg_namespace
+	)
+		AS pg_obj (objoid, classoid)
+	JOIN (
+			SELECT id, oid, type FROM mz_catalog.mz_objects
+			UNION ALL SELECT id, oid, 'schema' FROM mz_catalog.mz_schemas
```

Comment:
@jkosh44 what do you think about omitting schemas? That way the only objects present in this view use GlobalIds?

Reply:
It would certainly simplify this view, but it would be inconsistent with PostgreSQL.